flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer09.java

```diff
@@ -208,13 +208,12 @@ public FlinkKafkaConsumer09(List<String> topics, KeyedDeserializationSchema<T> d
 				if (partitionsForTopic != null) {
 					partitions.addAll(convertToFlinkKafkaTopicPartition(partitionsForTopic));
 				}
+				else{
+					LOG.info("Unable to retrieve any partitions for the requested topic: {}", topic);
+				}
 			}
 		}
 
-		if (partitions.isEmpty()) {
-			throw new RuntimeException("Unable to retrieve any partitions for the requested topics " + topics);
-		}
```

Comment:
I think Robert was suggesting to log messages only for cases where _some_ topics don't have partitions, so that the consumer doesn't just fail if for example only 1 out of 10 topics don't have partitions.

If ALL topics failed to return partitions in the end, we probably should still fail the consumer, like before.